### ISSUE TITLE
how to incorporate the regularization term into loss  computation in validation stage

### ISSUE THREAD

**Issue by danieljf24** [fchollet/keras#2104]
I designed a model with the loss function of  loss=f(W)+1/2||W||^2. I used the `train_on_batch` to train the model and calculate the loss in the training stage, and 'test_on_batch' to calculate the loss in the validation stage. I found that, the loss calculated by `train_on_batch` was f(W)+1/2||W||^2, but the loss only included f(W) when using 'test_on_batch', not incorporated the regularization term. It is reasonable in the majority cases.

But in my case, I want to calculate  f(W)+1/2||W||^2 and not train the model in validation stage.  In this case, `train_on_batch` and `test_on_batch` are not work. How can I rewrite the  `train_on_batch` or `test_on_batch` to meet my requirement? 

Thanks very much!

**Comment by NasenSpray** [comment]
You have to change `models.[Sequential or Graph].compile` so that regularizers are also added to `test_loss`.

**Comment by danieljf24** [comment]
thanks for your help, NasenSpray. But I still don't konw how to change the code, could me more suggestion? Which part in `models.[Sequential or Graph].compile` should I to change?

**Comment by NasenSpray** [comment]
After [here](https://github.com/fchollet/keras/blob/master/keras/models.py#L546-L547) and [here](https://github.com/fchollet/keras/blob/master/keras/models.py#L1297-L1298), insert
````
for r in self.regularizers:
    test_loss = r(test_loss)
````
